gcloud-java-storage/src/main/java/com/google/gcloud/storage/BlobWriteChannelImpl.java

```diff
@@ -48,22 +50,37 @@
   private boolean isOpen = true;
   private int chunkSize = DEFAULT_CHUNK_SIZE;
 
-  private transient StorageRpc storageRpc;
-  private transient StorageObject storageObject;
+  private final StorageRpc storageRpc;
+  private final StorageObject storageObject;
 
   BlobWriteChannelImpl(StorageOptions options, BlobInfo blobInfo,
       Map<StorageRpc.Option, ?> optionsMap) {
     this.options = options;
     this.blobInfo = blobInfo;
-    initTransients();
+    storageRpc = options.storageRpc();
+    storageObject = blobInfo.toPb();
     uploadId = storageRpc.open(storageObject, optionsMap);
   }
 
-  private void writeObject(ObjectOutputStream out) throws IOException {
+  BlobWriteChannelImpl(StorageOptions options, BlobInfo blobInfo, String uploadId) {
+    this.options = options;
+    this.blobInfo = blobInfo;
+    this.uploadId = uploadId;
+    storageRpc = options.storageRpc();
+    storageObject = blobInfo.toPb();
+  }
+
+  @Override
+  public RestorableState<BlobWriteChannel> save() {
     if (isOpen) {
       flush(true);
```

Comment:
I think now we can remove the "compact" param from the `flush` method and instead when we copy `buffer`
we can use `Arrays.copyOf(buffer, limit)` instead of `buffer.clone()`.

Also, I think "capturing" `limit` is not necessary as it is always going to be the length of the buffer and therefore
could be derived in the restore (that is true in either 'compact' mode or ,clearly, if we use the suggested copy).